<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Mono-dev] Socket.BeginReceive never throw Exception
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mono-devel-list%40lists.ximian.com?Subject=%5BMono-dev%5D%20Socket.BeginReceive%20never%20throw%20Exception&In-Reply-To=1270825257712-1819441.post%40n4.nabble.com">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="034734.html">
   <LINK REL="Next"  HREF="034744.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mono-dev] Socket.BeginReceive never throw Exception</H1>
    <B>Stefan [At]</B> 
    <A HREF="mailto:mono-devel-list%40lists.ximian.com?Subject=%5BMono-dev%5D%20Socket.BeginReceive%20never%20throw%20Exception&In-Reply-To=1270825257712-1819441.post%40n4.nabble.com"
       TITLE="[Mono-dev] Socket.BeginReceive never throw Exception">swieser1983 at gmail.com
       </A><BR>
    <I>Fri Apr  9 11:21:53 EDT 2010</I>
    <P><UL>
        <LI>Previous message: <A HREF="034734.html">[Mono-dev] switch gensources.sh in the 2.6 branch to /bin/sh (HEAD is /bin/sh already)
</A></li>
        <LI>Next message: <A HREF="034744.html">[Mono-dev] Socket.BeginReceive never throw Exception
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#34743">[ date ]</a>
              <a href="thread.html#34743">[ thread ]</a>
              <a href="subject.html#34743">[ subject ]</a>
              <a href="author.html#34743">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>
&gt;<i> socket.BeginReceive(receive_buffer, 0, FlockNode.buffer_size, 0, 
</I>&gt;<i> new AsyncCallback(ReceiveDone), socket);
</I>
&gt;<i> If the Socket gets disconnected by the remote host, 
</I>&gt;<i> ReceiveDone is called, and EndReceive will throw a SocketException; 
</I>&gt;<i> thus you know that your socket is now disconnected without having 
</I>&gt;<i> to poll it. With Mono, no exception is thrown. 
</I>
Just to be a bit more clear about this, not only no exception is thrown,
ReceiveDone isn't called either.
-- 
View this message in context: <A HREF="http://n4.nabble.com/Socket-BeginReceive-never-throw-Exception-tp1679973p1819481.html">http://n4.nabble.com/Socket-BeginReceive-never-throw-Exception-tp1679973p1819481.html</A>
Sent from the Mono - Dev mailing list archive at Nabble.com.
</PRE>




















<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="034734.html">[Mono-dev] switch gensources.sh in the 2.6 branch to /bin/sh (HEAD is /bin/sh already)
</A></li>
	<LI>Next message: <A HREF="034744.html">[Mono-dev] Socket.BeginReceive never throw Exception
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#34743">[ date ]</a>
              <a href="thread.html#34743">[ thread ]</a>
              <a href="subject.html#34743">[ subject ]</a>
              <a href="author.html#34743">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.ximian.com/mailman/listinfo/mono-devel-list">More information about the Mono-devel-list
mailing list</a><br>
</body></html>
